driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -165,7 +165,7 @@ public TimeoutSettings getTimeoutSettings() {
     }
 
     public long getMaxAwaitTimeMS() {
-        return hasTimeoutMS() ? 0 : timeoutSettings.getMaxAwaitTimeMS();
```

Comment:
In accordance with the CSOT specifications for the tailable-await cursor:

> 
"If the maxAwaitTimeMS option is set, drivers MUST use it as the maxTimeMS field on getMore commands."

The logic to determine whether `maxAwaitTimeMS `should be utilized is located in the AggregateOperationImpl and FindOperation.